### PR TITLE
Refactor Webpack Processes

### LINKED ISSUE #11: ONCE STABLE: Seperate webpacking draft files from webpacking target component files
Once we begin publishing lets separate the two

### LINKED ISSUE #75: Webpack builds twice on initial load
This is caused by the setup we have where we need to run webpack and then inject into the build process to build the master exports file. We need to find a way to get it to only build once.

### LINKED ISSUE #111: [iFrame] Each HMR reload above the iframe reloads the iframe itself, so HMR doesn't matter inside the iframe

### PR DESCRIPTION
**Pull request by zachintosh**
- Refactors the webpack process:
    - Demo amd Draft now have separate webpack configs
    - Developing react-draft locally runs webpack in a multicompiler mode that uses both configs
    - A config has been added for building a production version of react-draft
    - When consuming, not developing, react-draft, the demo config is used along with a production build of draft (previously it would transpile draft also)
    - webpack-dev-middleware and webpack-hot-middleware have been removed and webpack-dev-server is used in their place (much simpler)
- `npm run build` can be run in react-draft to build the production version
- Publishing to npm will only publish the launch file and the production build files (dist/)
- Switched to a light theme
- Moved the activity bar icons to the top, now displayed horizontally
- Renamed `master-exports` to `component-list`
- Added two more meta files like `component-list` named `component-meta` and `component-tree` (this lets Draft never need to import the actual components)
- Demo container now has a slighter shadow and squared corners
- New server side pretty logging implemented so the console isn't a garbage pile (import `log` from './lib/logger.js')

Known issues:

- The SelectedComponent reverts to the last one occasionally
- Running "npm run draft" doesn't work for some reason - you have to use node to run the launch file

TO DO BEFORE MERGING:
- Allow static site build for the example site (broke it with this update)
- Add docs on how to run dev mode and how to build prod version
- Fix tests
    

resolves #75
resolves #111 
resolves #11

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,12 +18,18 @@
     "type": "git",
     "url": "git+https://github.com/digital-taco/react-draft.git"
   },
+  "files": [
+    "dist/",
+    "launch.prod.js"
```
Comment (kyle-west) [inline package.json]:
```suggestion
    "bin/launch.prod.js"
```

File: package.json [hunk 2]
```diff
@@ -51,11 +62,11 @@
     "webpack": "^4.35.3",
     "webpack-cli": "^3.3.6",
     "webpack-dev-middleware": "^3.7.0",
-    "webpack-dev-server": "^3.7.2",
+    "webpack-dev-server": "^3.10.1",
     "webpack-hot-middleware": "^2.25.0"
   },
   "bin": {
-    "draft": "./bin/launch.js"
+    "draft": "./launch.prod.js"
```
Comment (kyle-west) [inline package.json]: Why did you move the path? Is it not still in `bin`? 
```suggestion
    "draft": "./bin/launch.prod.js"
```
I think this is what broke the build.

**Review by kyle-west — CHANGES_REQUESTED** [review]
I have 4 concerns:
- There is a LOT of commented out code that I would rather that we just delete it. If there is anything we should keep, then a comment as to why we need to keep it would be nice.
- From what you showed me yesterday, the color of the very small arrow on the left corner when all the tabs are closed was too low contrast. It needs to be more noticeable. Its the only GUI way of opening up the sidebar once everything is closed. I know the shortcuts work great, but my biggest worry is about discoverability with those. I think making that arrow more noticeable is important. 
- Something broke the built. I think it is the path of the launch file.
- It installs fine locally for me, but I get an error when I `npm link` it, so I could not test it out locally:
```
[Draft][Webpack] Building...

[Draft] Error:
 -------
./components/contexts/SelectedContext.js
Module not found: Error: Can't resolve '../../../out/component-meta' in '/Users/elderwestrrr/dev/digital-taco/react-draft/src/components/contexts'
```

Aside from that, what I you showed me yesterday looked really impressive. I am pretty happy with it and think it's ready for 1.0 release really soon.

File: package.json (hunk 2 above)
Comment (kyle-west) [reply package.json]: Should fix: 
> - Running "npm run draft" doesn't work for some reason - you have to use node to run the launch file

Comment (zachintosh) [reply package.json]: Good catch.

Comment (zachintosh) [reply package.json]: The error received is that it can't find the command "draft", rather than it not being able to find the file. So this doesn't fix the error, but it does need to be fixed.

**Comment by zachintosh** [comment]
> From what you showed me yesterday, the color of the very small arrow on the left corner when all the tabs are closed was too low contrast. It needs to be more noticeable. Its the only GUI way of opening up the sidebar once everything is closed. I know the shortcuts work great, but my biggest worry is about discoverability with those. I think making that arrow more noticeable is important.

I agree.

**Comment by zachintosh** [comment]
The build error you experienced is something I discovered last night. It's because it's bundling the component meta files, which I hadn't thought of. To get it to work, link it to sandbox and run `launch.dev.js`. That'll generate the files, which then get bundled. I'm working on making it so they aren't bundled.

EDIT: Just saw that it happened on linking. I'll fix it.

File: bin/launch.prod.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    // Add any custom middleware
+    draftConfig.middleware && draftConfig.middleware(app)
+  },
+}
+
+const server = new WebpackDevServer(demoCompiler, devServerOptions)
+
+function launchServer() {
+  server.listen(port, '127.0.0.1', () => log(`Server launched on port ${port}`, 'Server'))
```
Comment (kyle-west) [inline bin/launch.prod.js]: Is there a reason why we use `127.0.0.1` and not `localhost`? Not that it matters.

File: config/demo.webpack.js [hunk 4]
```diff
@@ -1,109 +1,107 @@
...
     resolve: {
       symlinks: true,
-      alias: draftConfig.DEBUG
-        ? {
-            // Resolve the path to React so we don't import multiple react versions
-            react: path.resolve(__dirname, './node_modules/react'),
-            '@emotion/core': path.resolve(__dirname, './node_modules/@emotion/core'),
-          }
-        : undefined,
+      alias: {
```
Comment (kyle-west) [inline config/demo.webpack.js:57]: Are we confident that this works in a non-linked environment (a.k.a a true `npm install`)? I see that the `draftConfig.DEBUG` conditional logic was removed. Or is this still handled by the nature of having the two separate config files?

File: lib/logger.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+const colors = require('colors')
+
+function log(str, tags = []) {
```
Comment (kyle-west) [inline lib/logger.js:3]: This is wonderful. The console looks beautiful, and you know that I mean it when I saw I appreciate a beautiful CLI.

File: package.json [hunk 6]
```diff
@@ -1,12 +1,14 @@
 {
   "name": "@digital-taco/react-draft",
-  "version": "0.4.3",
+  "version": "0.5.0",
   "description": "Develop your React components in isolation without any configuration",
-  "main": "index.js",
+  "main": "launch.prod.js",
```
Comment (kyle-west) [inline package.json]: Does this need to change too?
```suggestion
  "main": "bin/launch.prod.js",
```

**Review by kyle-west — COMMENTED** [review]
I think this looks so great! Awesome work. I have a few comments but nothing to stop us from merging this PR.

**Comment by kyle-west** [comment]
Couple more (non-blocking) issues:
- I ran it on react-draft-sandbox and it no longer opens a new chrome window for me automatically.
- I would like the place holder  text to be greyed out more. I thought it was real code for a second when I opened it up:

Screen Shot 2020-01-28 at 7 18 53 PM

File: bin/launch.prod.js (hunk 3 above)
Comment (zachintosh) [reply bin/launch.prod.js]: Beats me
